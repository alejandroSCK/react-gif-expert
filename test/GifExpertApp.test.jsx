import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { GifExpertApp } from "../src/GitExpertApp";

describe('GifExpertApp test', () => {
    test('snapshot initial state', () => {
        const { container } = render(<GifExpertApp/>);
        expect(container).toMatchSnapshot();
    });

    test('Should add new category', () => {
        const inputValue = 'Dark Souls';

        render(<GifExpertApp/>);
        
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, { target: { value: inputValue}});
        fireEvent.submit(form);

        expect( screen.getByText(inputValue) ).toBeTruthy();
    });

    test('should show image', async() => {
        render(<GifExpertApp/>);

        await waitFor(() => expect(screen.getAllByRole('img').length).toBeGreaterThan(0));

        expect(screen.getAllByRole('img').length).toBeGreaterThan(0);
    });
});