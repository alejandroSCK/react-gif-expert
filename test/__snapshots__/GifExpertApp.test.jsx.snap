// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp test snapshot initial state 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
